front/app/modules/commercial/moderation/admin/containers/index.tsx

```diff
@@ -313,53 +316,55 @@ const Moderation = memo<Props & InjectedIntlProps>(({ className, intl }) => {
-        const newSelectedRows = includes(selectedRows, selectedModerationId)
-          ? selectedRows.filter((id) => id !== selectedModerationId)
-          : [...selectedRows, selectedModerationId];
-        setSelectedRows(newSelectedRows);
+        const newSelectedRows = includes(
+          selectedRowModerationIds,
+          selectedModerationId
+        )
+          ? selectedRowModerationIds.filter((id) => id !== selectedModerationId)
+          : [...selectedRowModerationIds, selectedModerationId];
+        setSelectedRowModerationIds(newSelectedRows);
       }
     },
-    [selectedRows, processing]
+    [selectedRowModerationIds, processing]
   );
 
-  const removeFlags = () => {};
-  // const removeFlags = useCallback(
-  //   async (event: React.FormEvent) => {
-  //     if (
-  //       selectedRowsWithContentWarning.length > 0 &&
-  //       !isNilOrError(moderationItems) &&
-  //       moderationStatus &&
-  //       !processing
-  //     ) {
-  //       event.preventDefault();
-  //       trackEventByName(
-  //         removeFlagButtonClicked
-  //         { selectedItemsCount: selectedRows.length }
-  //       );
-  //       setProcessing(true);
-  //       const moderations = selectedRowsWithContentWarning.map((moderationId) =>
-  //         moderationItems.find((item) => item.id === moderationId)
-  //       ) as IModerationData[];
-  //       const updatedFlaggedStatus = 'approved';
-  //       const promises = moderations.map((moderation) =>
-  //         updateModerationFlaggedStatus(
-  //           moderation.id,
-  //           moderation.attributes.moderatable_type,
-  //           updatedFlaggedStatus
-  //         )
-  //       );
-  //       await Promise.all(promises);
-  //       setProcessing(false);
-  //       setSelectedRows([]);
-  //     }
-  //   },
-  //   [selectedRows, moderationItems, moderationStatus]
-  // );
+  const removeFlags = useCallback(async () => {
+    if (
+      selectedRowModerationIds.length > 0 &&
+      !isNilOrError(moderationItems) &&
+      !processing
+    ) {
+      const selectedModerationItems = moderationItems.filter((modItem) =>
+        selectedRowModerationIds.includes(modItem.id)
+      );
+
+      const promises = selectedModerationItems.map((moderationWithWarning) => {
+        const flagId =
+          moderationWithWarning.relationships.inappropriate_content_flag?.data
+            .id;
+
+        // Moderation items having a content flag relationship doesn't mean
+        // the flag is currently on (see the service file for more info).
+        // I think, however, in this case having the relationship is a decent proxy
+        // and the code is simpler than somehow fetching the flags of all selected items,
+        // then checking whether their flag is really turned on
+        // by checking the reason_code and toxicity_label, and only then removing the flag.
+        if (flagId) {
+          return removeInappropriateContentFlag(flagId);
```

Comment:
So here I'm not 100% sure what is best. We might make a few extra requests for items that have a flag which is already turned off. (If you're not following: if a content flag is removed, it's not really removed but rather marked as deleted by changing the reason_code attribute to null. This way, we keep the flag in case we want to re-enable it again).

The other option is adding a lot of extra code to grab the flags, check their reason_code and manage all that in this component, so I think this is a good option.